### PR TITLE
4039 5 reactions upgrade react and styled components

### PR DESCRIPTION
**Pull request by StevenZuber**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -146,13 +146,14 @@
     "postcss-load-config": "1.1.0",
     "postcss-loader": "1.2.2",
     "raw-loader": "~0.5.1",
-    "react": "16.2.0",
+    "react": "^16.11.0",
+    "react-codemod": "^5.1.1",
```
Comment (samageloff) [inline package.json]: let's remove `react-codemod` as this'll likely be the last time we'll need it. (note: re-run `npm install` so the package.lock is updated)

File: package.json [hunk 2]
```diff
@@ -146,13 +146,14 @@
...
     "react-dnd-test-backend": "~1.0.2",
     "react-docgen": "2.20.0",
-    "react-dom": "16.2.1",
+    "react-dom": "^16.11.0",
```
Comment (samageloff) [inline package.json]: use exact version eg: `16.11.0`

File: package.json [hunk 3]
```diff
@@ -146,13 +146,14 @@
...
     "react-highlight": "~0.8.0",
     "react-svg-inline": "~1.1.0",
-    "react-test-renderer": "16.2.0",
+    "react-test-renderer": "^16.11.0",
```
Comment (samageloff) [inline package.json]: ditto above

File: package.json [hunk 4]
```diff
@@ -170,7 +171,7 @@
   },
   "peerDependencies": {
     "styled-components": ">= 2.0.0",
-    "react": ">= 15.3.0 <= 16.2.0",
-    "react-dom": ">= 15.3.0 <= 16.2.0"
+    "react": ">= 15.3.0 <= 16.9.0",
```
Comment (samageloff) [inline package.json]: if we're upgrading to `16.11.0`, we can bump the peer dependency as well

File: package.json [hunk 5]
```diff
@@ -170,7 +171,7 @@
...
+    "react-dom": ">= 15.3.0 <= 16.9.0"
```
Comment (samageloff) [inline package.json]: ditto above

File: src/components/Spinner/Spinner.js [hunk 6]
```diff
@@ -125,8 +125,8 @@ class Spinner extends React.Component {
   }
 
   render = () => {
-    const { backgroundColor, className, color, optClass, position, size, type } = this.props
-    const { loading } = this.state
+    const { backgroundColor, className, color, optClass, position, size, type } = this.props;
```
Comment (samageloff) [inline src/components/Spinner/Spinner.js]: remove `;`

File: src/components/Spinner/Spinner.js [hunk 7]
```diff
@@ -125,8 +125,8 @@ class Spinner extends React.Component {
...
+    const { loading } = this.state;
```
Comment (samageloff) [inline src/components/Spinner/Spinner.js]: ditto above

**Comment by jpclark6** [comment]
@samageloff made the updates if you'd like to re-check and then let us know if things look 👍

**Pull request merged**